### PR TITLE
GH-12630 Migrate tests from model/file_info_test.go to use testify 

### LINKED ISSUE #12630: Migrate tests from "model/file_info_test.go" to use testify
Mattermost is migrating its server tests from using the `t.Fatal` method calls on their checks to use the [testify toolkit](https://github.com/stretchr/testify). The goal of this migration is to improve the readability of the tests, making them more concise and meaningful. This Help Wanted issue is to migrate the tests in the `model/file_info_test.go` file.

The expected way to migrate this is to go to the `model/file_info_test.go` file and modify the calls to `t.Fatal` and the condition checks that lead to them, replacing them with calls to the [the `require` package](https://godoc.org/github.com/stretchr/testify/require) if a failure should stop the execution of the test, or [the `assert` package](https://godoc.org/github.com/stretchr/testify/assert) if it shouldn't.

Example: mattermost/mattermost-server#12039

If you have questions about the ticket or you need any help, feel free to contact `miguel.delacruz` or `jesus.espino` in <https://community.mattermost.com/>

----
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

JIRA: https://mattermost.atlassian.net/browse/MM-19240

### PR DESCRIPTION
**Pull request by dedifferentiator**
#### Summary
Migrate tests from `model/file_info_test.go` to use testify 

#### Ticket Link
Fixes #12630

### CODE REVIEW

File: model/file_info_test.go [hunk 1]
```diff
@@ -22,179 +24,127 @@ func TestFileInfoIsValid(t *testing.T) {
 		Path:      "fake/path.png",
 	}
 
-	if err := info.IsValid(); err != nil {
-		t.Fatal(err)
-	}
+	err := info.IsValid()
+	require.Nil(t, err)
```
Comment (Willyfrog) [inline model/file_info_test.go]: why the 2-step process? whouldn't it be simpler as
```suggestion
	require.Nil(t, info.IsValid())
```
I see you are consistent in this two step process, but we are not really caring about the err value other than checking it's not nil, so I'd suggest to change it elsewhere

File: model/file_info_test.go [hunk 2]
```diff
@@ -22,179 +24,127 @@ func TestFileInfoIsValid(t *testing.T) {
...
 
 func TestFileInfoIsImage(t *testing.T) {
 	info := &FileInfo{
 		MimeType: "image/png",
 	}
 
-	if !info.IsImage() {
-		t.Fatal("file is an image")
-	}
+	assert.False(t, !info.IsImage(), "file is an image")
```
Comment (Willyfrog) [inline model/file_info_test.go]: double negation? why not use assert.True?

Comment (dedifferentiator) [reply model/file_info_test.go]: Fair enough

File: model/file_info_test.go (hunk 1 above)
Comment (dedifferentiator) [reply model/file_info_test.go]: Thanks, it also was bothering me, so I decided to make it like more explicit, I don't know. Someone should just say that :)

File: model/file_info_test.go [hunk 3]
```diff
@@ -22,179 +24,117 @@ func TestFileInfoIsValid(t *testing.T) {
...
-	} else if info.Extension != "tif" {
-		t.Fatalf("Got incorrect extension: %v", info.Extension)
-	} else if info.MimeType != "image/tiff" && info.MimeType != "image/x-tiff" {
-		t.Fatalf("Got incorrect mime type: %v", info.MimeType)
+	info, err = GetInfoForBytes("file.tif", fakeFile)
+	require.Nil(t, err)
+	assert.Equalf(t, info.Name, "file.tif", "Got incorrect filename: %v", info.Name)
+	assert.Equalf(t, info.Extension, "tif", "Got incorrect extension: %v", info.Extension)
+	if info.MimeType != "image/tiff" {
+		assert.Equalf(t, info.MimeType, "image/x-tiff", "Got incorrect mime type: %v", info.MimeType)
```
[the remark this answers is not in the document]
Comment (dedifferentiator) [reply model/file_info_test.go]: By the way tests won't pass if it's not x-tiff or tiff. 1.if it's not tiff it asserts equal x-tiff; 2. if it's tiff it passes which is correct.

[the remark this answers is not in the document]
Comment (Willyfrog) [reply model/file_info_test.go]: there is also [Condition](https://godoc.org/github.com/stretchr/testify/assert#Condition) for complex comparisons.
another option is use `assert.True` or `assert.False` with the condition inside

[the remark this answers is not in the document]
Comment (hanzei) [reply model/file_info_test.go]: @dedifferentiator Do you have questions about this feedback?

[the remark this answers is not in the document]
Comment (dedifferentiator) [reply model/file_info_test.go]: @hanzei not yet, I've got sick, but I think I'll get to this in a day

**Comment by Willyfrog** [comment]
looking good, can you get it up to date?

**Comment by dedifferentiator** [comment]
@Willyfrog Done

**Pull request merged**